sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java

```diff
@@ -603,6 +625,36 @@ private boolean getSuccess(SpanData span) {
 
     @Nullable
     public static String getHttpUrlFromServerSpan(Attributes attributes) {
+        String httpUrl = getHttpUrlFromServerSpanStableSemconv(attributes);
+        if (httpUrl != null) {
+            return httpUrl;
+        }
+        return getHttpUrlFromServerSpanOldSemconv(attributes);
+    }
+
+    @Nullable
+    private static String getHttpUrlFromServerSpanStableSemconv(Attributes attributes) {
+        String scheme = attributes.get(SemanticAttributes.URL_SCHEME);
+        if (scheme == null) {
+            return null;
+        }
+        String path = attributes.get(SemanticAttributes.URL_PATH);
+        if (path == null) {
+            return null;
+        }
+        String host = attributes.get(SemanticAttributes.SERVER_ADDRESS);
+        if (host == null) {
+            return null;
+        }
+        Long port = attributes.get(SemanticAttributes.SERVER_PORT);
```

Comment:
Perhaps the addition of a `getNotNullAttribute(AttributeKey... attributeKeys)` method returning the the first non-null attribute could simplify the code in several places. This method may belong to the `SemanticAttributes` class.

Reply:
it's not clear to me how this would work here, can you make a (full) github "suggestion" so I can understand better?

Reply:
Sorry, I had read the code too fast. The logic is quite different between the methods.